Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/lucene/impl/LuceneUtility.java

```diff
@@ -625,12 +628,17 @@ private void buildQuery(BooleanClause.Occur occur, PerFieldAnalyzerWrapper perFi
     }
 
     private void buildTokenizedOrWildCardQuery(LuceneField luceneFieldAnnotation, String searchString, QueryBuilder qBuild,
-            BooleanQuery.Builder tokenizedQueryBuilder) {       
+            BooleanQuery.Builder tokenizedQueryBuilder, boolean isSearchOnDescription) {       
             if (luceneFieldAnnotation.isWildCardSearchEnabled()) {
                 BooleanQuery.Builder analyzedBuilder = new BooleanQuery.Builder();             
                 createWildCardSearchQuery(luceneFieldAnnotation, searchString, analyzedBuilder, Occur.SHOULD);
                 analyzedBuilder.add(new BooleanClause(
-                        (qBuild.createPhraseQuery(luceneFieldAnnotation.lucenefieldName(), searchString)), Occur.SHOULD));
+                        (qBuild.createBooleanQuery(luceneFieldAnnotation.lucenefieldName(), searchString)), Occur.SHOULD));
+                if(isSearchOnDescription) {
+                    analyzedBuilder.add(new BooleanClause(
+                        (qBuild.createBooleanQuery(SearchFieldNames.DESCRIPTION, searchString)), Occur.SHOULD));
+                }
```

Comment:
why is this added here? this would be added multiple times (for every field that is wild card search enabled). There is a similar block in `buildQuery` that is not in develop but doesn't show up as changed. What's up with that?

Reply:
The reason is that because the word and description query need to be combined as one.  We are adding Word and description queries to the `analyzedBuilder`  BooleanQuery.Builder object. And you are right, it will be added for every wild card search, but it will work in our case because only the word is enabled for a wildcard search.


Ah, I'm merging the current branch to the CCP-226, not to the develop. As I told you in the above comment, I messed up while pushing commits. It got mixed between CCP-226 and CCP-232. That's the reason why it's not showing up here.

Reply:
I think we need to make some changes to the search. This doesn't make sense and would create odd queries when used e.g. by the rest endpoint search. There are fields that make sense to search and fields that are limiting the search. I would add another search method (leave the current one the way it was (except refactoring) so that the rest search still works). The new search method takes POS and concept list as arguments (each could be more than just one value). If those are set than the query should have a `AND conceptlist in [...]`/`AND POS in [...]`. All fields that are actually search should be combined with the given operator (in case of the front end search `OR`). Does this make sense? It should be easy then also to add a search to the concept lists.